java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -86,7 +87,8 @@
       try {
         evaluateStatement(ImmutableList.of(state), tree.block());
       } catch (SymbolicExecutionException e) {
-        LOGGER.info("analysis of " + tree + " failed: " + e.getMessage(), e);
+        JavaTree javaTree=(JavaTree)tree;
+        LOGGER.info("in " + javaTree.getToken().getURI() + ": analysis of " + tree.simpleName().name() + " at line "+javaTree.getLine() + " failed: " + e.getMessage());
```

Comment:
What about JavaFileScannerContext.getFile() then?
Though this would require to propagate the file to the engine through the check.